src/containers/FilterBarContainer.tsx

```diff
@@ -0,0 +1,50 @@
+import { connect } from "react-redux";
+import * as React from "react";
+import { Store, asyncActions, facetsActions, searchParameterActions } from "azsearchstore";
+import * as redux from "redux";
+import FilterBar from "../components/FilterBar";
+
+function getReturnType<RT>(expression: (...params: any[]) => RT): RT {
+    return {} as RT;
+}
+
+export interface OwnProps {
+}
+
+const mapDispatchToProps = (dispatch: redux.Dispatch<any>, ownProps: OwnProps) => {
+  return {
+    onClear: () => {
+      dispatch(facetsActions.clearFacetsSelections);
+      dispatch(searchParameterActions.setPage(1));
+      dispatch(asyncActions.fetchSearchResults);
+    }
+  };
+};
+
+function mapStateToProps(state: Store.SearchState, ownProps: OwnProps) {
+  return {
+    hasSelectedFacets: checkForAppliedFacets(state.facets.facets)
+  };
+}
+
+function checkForAppliedFacets(facets: { [key: string]: Store.Facet }) {
+  for (let facetKey in facets) {
+    let checkboxFacet = facets[facetKey] as Store.CheckboxFacet;
```

Comment:
I think that given that this is computed data, this is probably the appropriate place for it, although I could see this being a good general purpose property we add to the store that gets updated when facets are cleared or selected. Using the filter string seems pretty safe.